hydromt_sfincs/sfincs.py

```diff
@@ -1011,6 +1016,126 @@ def setup_cn_infiltration(self, cn, antecedent_moisture="avg", reproj_method="me
+        """
+
+        # Read the datafiles
+        data_catalog = hydromt.DataCatalog(data_libs=data_catalog)
+        da_landuse = data_catalog.get_rasterdataset(
+            name_files[0], geom=self.region, buffer=10
+        )  # landcover
+        da_HSG = data_catalog.get_rasterdataset(
+            name_files[1], geom=self.region, buffer=10
+        )  # HSG
+        da_Ksat = data_catalog.get_rasterdataset(
+            name_files[2], geom=self.region, buffer=10
+        )  # Ksat
+        df_map = self.data_catalog.get_dataframe(reclass_table, index_col=0)
+
+        # Define outputs
+        da_smax = xr.full_like(self.mask, -9999, dtype=np.float32)
+        da_kr = xr.full_like(self.mask, -9999, dtype=np.float32)
+
+        # Define the blocks
+        nrmax = block_size
+        nmax = np.shape(self.mask)[0]
+        mmax = np.shape(self.mask)[1]
+        grid_dim = (nmax, mmax)
+        dx = self.config["dx"]
+        refi = dx / 30  # finest resolution of NLCD is ~30m
+        nrcb = int(np.floor(nrmax / refi))  # nr of regular cells in a block
+        nrbn = int(np.ceil(nmax / nrcb))  # nr of blocks in n direction
+        nrbm = int(np.ceil(mmax / nrcb))  # nr of blocks in m direction
+        x_dim, y_dim = self.mask.raster.x_dim, self.mask.raster.y_dim
+
+        # avoid blocks with width or height of 1
+        merge_last_col = False
+        merge_last_row = False
+        if mmax % nrcb == 1:
+            nrbm -= 1
+            merge_last_col = True
+        if nmax % nrcb == 1:
+            nrbn -= 1
+            merge_last_row = True
+
+        ## Loop through blocks
+        ib = -1
+        for ii in range(nrbm):
+            bm0 = ii * nrcb  # Index of first m in block
+            bm1 = min(bm0 + nrcb, mmax)  # last m in block
+            if merge_last_col and ii == (nrbm - 1):
+                bm1 += 1
+
+            for jj in range(nrbn):
+                bn0 = jj * nrcb  # Index of first n in block
+                bn1 = min(bn0 + nrcb, nmax)  # last n in block
+                if merge_last_row and jj == (nrbn - 1):
+                    bn1 += 1
+
+                # Count
+                ib += 1
+                # Get some prints => no idea how to get the logger info
+                print(
+                    f"\nblock {ib + 1}/{nrbn * nrbm} -- "
+                    f"col {bm0}:{bm1-1} | row {bn0}:{bn1-1}"
+                )
+
+                # calculate transform and shape of block at cell and subgrid level
+                da_mask_block = self.mask.isel(
+                    {x_dim: slice(bm0, bm1), y_dim: slice(bn0, bn1)}
+                ).load()
+
+                # Call workflow
+                [
+                    da_smax_block,
+                    da_kr_block,
+                ] = workflows.merge.curvenumber_recovery_determination(
+                    da_landuse, da_HSG, da_Ksat, df_map, da_mask_block
+                )
```

Comment:
the square brackets here are not necessary and potentially make it harder to read?

```suggestion
                # Call workflow
                da_smax_block, da_kr_block = workflows.merge.curvenumber_recovery_determination(
                    da_landuse, da_HSG, da_Ksat, df_map, da_mask_block
                )
```